### PR TITLE
fix(frontend/controller): convert search query to ascii when required

### LINKED ISSUE #166: Unify accent insertion/deletion in search bar
Issues with inserting accent chars. My first guess would be the interpreter for unicode/ASCII in search causes accents to be placed differently. Backspace is not following the interpreter rule. So either use pure ASCII convention or pure Unicode convention for adding/removing accents.

Example: `#bi<backspace>`
Example: `#pRb vis` you'll see the `i` char go behind the `v` and then after typing the `s` it will go back even another char. Then try adding more chars or backspacing from here. It will keep going back another char.
Example: `#hm qy ki` (unicode style) or `#hm qy ik` (ascii style) should be showing the result `ਹਮ ਤੇ ਕਿਛੂ ਨ ਹੋਇ ਮੇਰੇ ਸ੍ਵਾਮੀ`

_Originally posted by @bhajneet in https://github.com/ShabadOS/desktop/pull/161#issuecomment-500635936_

Child of #211

### PR DESCRIPTION
**Pull request by bhajneet**
otherwise it will unnecessarily move around sihari accent character (ASCII i)

fix #166

---

## Summary

The search query was unnecessarily being converted to ascii which caused it to move the accent character (sihari). Making sure the query is not ascii before converting it resolved it.

### Before
before

### After
after

## Tests

Tried ascii and unicode on desktop and mobile phone.

Tried all sorts of various characters on keyboard without having the render mess up (and hopefully assuming that applies to query as well, no easy way to test all those manually)

## Other

I am unsure if there is a way to refactor such that I don't have to use the same conditional in two separate locations

## PM / Time

Took between 30-45 minutes

### CODE REVIEW

File: app/frontend/src/Controller/Search.js [hunk 1]
```diff
@@ -44,7 +44,9 @@ const getSearchParams = searchQuery => {
   // Extract anchors and search query
   const [ , anchor, query ] = searchQuery.match( searchRegex )
 
-  const inputValue = toAscii( query )
+  let hasMoreThanAscii = [...query].some(char => char.charCodeAt(0) > 127);
```
Comment (Harjot1Singh) [inline app/frontend/src/Controller/Search.js:47]: Use `const` over `let` whenever that value never changes. It helps the reader understand that there's not going to be a reassignment later to watch out for.

I see that you're using `[...query]` to coerce to an array, so that you can gain access to `Array.some`, clever trick. I was going to suggest only checking the last character, but am unsure if this would break pasting into the search box. Worth a try, since it'll mean performance is independent of the length of the search query.

Is `127` the right value to use? Aren't there some chars in the "gurmukhi-ascii" range that exceed this?

File: app/frontend/src/lib/controller.js [hunk 2]
```diff
@@ -107,7 +107,7 @@ class Controller extends EventEmitter {
    * @param type The type of search (first-letter/full-word).
    * @param options Additional options to pass.
    */
-  search = ( query, type, options = {} ) => this.sendJSON( `search:${type}`, { ...options, query: toAscii( query ) } )
+  search = ( query, type, options = {} ) => this.sendJSON( `search:${type}`, { ...options, query: [...query].some(char => char.charCodeAt(0) > 127) ? toAscii( query ) : query } )
```
Comment (Harjot1Singh) [inline app/frontend/src/lib/controller.js:110]: Why is this required? Does it not search correctly if this is the case?

Comment (bhajneet) [reply app/frontend/src/lib/controller.js:110]: Yes, it fails, see edited part of #540 

I think: (1) do the same thing in both places, or (2) pass query around properly?

File: app/frontend/src/Controller/Search.js (hunk 1 above)
Comment (bhajneet) [reply app/frontend/src/Controller/Search.js:47]: 1. Will change to const
2. Can try checking only last char entered, but need to test if I type fast enough or what
3. 127 is not really important, this only breaks on one character as far as I can tell, so just checking for that one would be good enough too I think. I do believe ASCII range is 128 though, so in that sense it should be correct, but you're right that gurmukhi-ascii range is definitely exceeding this. I think I will change it to look only for the character causing the issue

Comment (Harjot1Singh) [reply app/frontend/src/Controller/Search.js:47]: I also did not think about how it'd affect deletion of characters

Comment (Harjot1Singh) [reply app/frontend/src/Controller/Search.js:47]: I also did not think about how it'd affect deletion of characters

Comment (bhajneet) [reply app/frontend/src/Controller/Search.js:47]: doesn't work also because some chars in unicode are combo chars, typing them on keyboard requires you type the base then the accent on it (ਸ +  ਿ = ਸਿ), and querying on last char vs whole string messes that up

side note, converting from unicode to ascii also hides all "suggestions" in mobile keyboards, not sure what to do about that, my initial thought:

- Don't convert query rendered in input, only convert in db search (so this removes the line 47 but keeps the line 110

I can't imagine we need to support both unicode and ascii at the same time, so it should be okay?

Comment (bhajneet) [reply app/frontend/src/Controller/Search.js:47]: Nvm, have to look at the functions more closely...

**Comment by Sarabveer** [comment]
https://github.com/ShabadOS/database/issues/1333

**Comment by Sarabveer** [comment]
This is wrong, GurbaniAkhar has characters outside of the 128-char ASCII range.

It should be solved in https://github.com/ShabadOS/database/issues/1333

**Comment by bhajneet** [comment]
Closing in favor of using isGurmukhi gurmukhi utils https://github.com/ShabadOS/gurmukhi-utils/pull/141
